scripts/readrecording.js

```diff
@@ -71,169 +71,170 @@ example['responses'] = {}
 
 var accErrors = {}
 
-var getFileList = function(dir, filelist) {
-    var files = fs.readdirSync(dir);
-    filelist = filelist || [];
-    files.forEach(function(file) {
-        if (fs.statSync(pathlib.join(dir, file)).isDirectory()) {
-            filelist = getFileList(pathlib.join(dir, file), filelist);
-        }
-        else {
-            filelist.push(pathlib.join(dir, file));
-        }
-    });
-    return filelist;
+var getFileList = function (dir, filelist) {
+  var files = fs.readdirSync(dir);
+  filelist = filelist || [];
+  files.forEach(function (file) {
+    if (fs.statSync(pathlib.join(dir, file)).isDirectory()) {
+      filelist = getFileList(pathlib.join(dir, file), filelist);
+    }
+    else {
+      filelist.push(pathlib.join(dir, file));
+    }
+  });
+  return filelist;
 };
-var filesArray =[]
+var filesArray = []
 getFileList(pathToRecordings, filesArray)
 
-recordingFiles = filesArray 
+recordingFiles = filesArray
 
 var example = {};
 parser.parse(swaggerObject)
-.then(function(api){
+  .then(function (api) {
     console.log("API name: " + api.info.title)
     specApiVersion = api.info.version
     paths = api.paths
-    
-    for (var recordingFile in recordingFiles){
-        console.log("*****************")
-        console.log("Recording file: " + recordingFiles[recordingFile])
-        recordingFileName = recordingFiles[recordingFile]
-        try{
-            var recording = JSON.parse(fs.readFileSync( recordingFileName)); 
-
-            pathIndex = 0
-            var pathParams = {}
-            for (var path in paths){
-                pathIndex ++;
-                searchResult = path.match(/\/{\w*\}/g)
-                pathParts = path.split('/')
-                pathToMatch = path
-                pathParams = {}
-                for (var sr in searchResult){
-                    match = searchResult[sr]
-                    splitRegEx = /[{}]/
-                    pathParam = match.split(splitRegEx)[1]
-                    
-                    for (var part in pathParts){
-                        pathPart = "/" + pathParts[part]
-                        if (pathPart.localeCompare(match) == 0){
-                            pathParams[pathParam] = part
-                        }
-                    } 
-                    pathToMatch = pathToMatch.replace(match,"/[^\/]+");
+
```

Comment:
Actually i was debugging to understand the logic and silly javascript works without `;` when written outside of class. So i think i just added bunch of `;` and `let` in this file